src/DistortableImageOverlay.js

```diff
@@ -245,6 +247,115 @@ L.DistortableImageOverlay = L.ImageOverlay.extend({
+        // ((c1PY > c2PY && c1PY > c3PY) && (c0PY > c2PY && c0PY > c3PY))
+      );
+	},
+
+	_isRightSideUp: function () {
+		var c0P = this._map.latLngToLayerPoint(this._corners[0]);
+    var c1P = this._map.latLngToLayerPoint(this._corners[1]);
+    var c2P = this._map.latLngToLayerPoint(this._corners[2]);
+    var c3P = this._map.latLngToLayerPoint(this._corners[3]);
+    var c0PY = L.Util.formatNum(c0P.y, 2);
+    // var c0Px = L.Util.formatNum(c0P.x, 2);
+    var c1PY = L.Util.formatNum(c1P.y, 2);
+    // var c1PX = L.Util.formatNum(c1P.x, 2);
+    var c2PY = L.Util.formatNum(c2P.y, 2);
+    var c3PY = L.Util.formatNum(c3P.y, 2);
+    // var c3PX = L.Util.formatNum(c3P.x, 2);
+    // var c2PX = L.Util.formatNum(c2P.x, 2);
+    // return ((c1PY > c2PY && c0PY > c3PY) && (c1PX > c2PX && c1PX > c3PX));
+    return c0PY < c2PY && c0PY < c3PY && (c1PY < c2PY && c1PY < c3PY);
+
+	},
+
+	_isUpsideDown: function () {
+		window.thisss = this;
+		var c0P = this._map.latLngToLayerPoint(this._corners[0]);
+		var c1P = this._map.latLngToLayerPoint(this._corners[1]);
+		var c2P = this._map.latLngToLayerPoint(this._corners[2]);
+		var c3P = this._map.latLngToLayerPoint(this._corners[3]);
+		var c0PY = L.Util.formatNum(c0P.y, 2);
+		// var c0Px = L.Util.formatNum(c0P.x, 2);
+		var c1PY = L.Util.formatNum(c1P.y, 2);
+		// var c1PX = L.Util.formatNum(c1P.x, 2);
+		var c2PY = L.Util.formatNum(c2P.y, 2);
+		var c3PY = L.Util.formatNum(c3P.y, 2);
+		// var c3PX = L.Util.formatNum(c3P.x, 2);
+		// var c2PX = L.Util.formatNum(c2P.x, 2);
+		// return ((c1PY > c2PY && c0PY > c3PY) && (c1PX > c2PX && c1PX > c3PX));
+		return ((c0PY > c2PY && c0PY > c3PY) && (c1PY > c2PY && c1PY > c3PY));
+
+	},
+
+	_isSidewaysLeft: function () {
+			var c0P = this._map.latLngToLayerPoint(this._corners[0]);
+      var c1P = this._map.latLngToLayerPoint(this._corners[1]);
+      var c2P = this._map.latLngToLayerPoint(this._corners[2]);
+      var c3P = this._map.latLngToLayerPoint(this._corners[3]);
+      window.c0P = c0P;
+      window.c1P = c1P;
+      window.c2P = c2P;
+      window.c3P = c3P;
+      var c0PY = L.Util.formatNum(c0P.y, 2);
+      var c0PX = L.Util.formatNum(c0P.x, 2);
+      var c1PY = L.Util.formatNum(c1P.y, 2);
+      var c1PX = L.Util.formatNum(c1P.x, 2);
+      // var c2PY = L.Util.formatNum(c2P.y, 2);
+      var c3PY = L.Util.formatNum(c3P.y, 2);
+      var c3PX = L.Util.formatNum(c3P.x, 2);
+      var c2PX = L.Util.formatNum(c2P.x, 2);
+      return ((c0PY < c1PY && c0PY < c3PY) || (c0PY > c1PY && c0PY > c3PY)) && (c1PX < c3PX && c0PX < c2PX);
+	},
+
+	_isSidewaysRight: function () {
+		var c0P = this._map.latLngToLayerPoint(this._corners[0]);
+		var c1P = this._map.latLngToLayerPoint(this._corners[1]);
+		var c2P = this._map.latLngToLayerPoint(this._corners[2]);
+		var c3P = this._map.latLngToLayerPoint(this._corners[3]);
+		window.c0P =c0P;
+		window.c1P = c1P;
+		window.c2P = c2P;
+		window.c3P = c3P;
```

Comment:
Are we using these `window` fields anywhere?

Reply:
Nope! Will be gone in next commit!